DOAJ/doaj
# Feature/3190 remove esprit

# Remove esprit

See https://github.com/DOAJ/doajPM/issues/3190

Remove esprit from DOAJ project (apart from in legacy scripts which will no longer run) in favour of the Elasticsearch python library and our `es_connection` singleton.

## Categorisation

This PR...
- [ ] affects the monitoring

## Basic PR Checklist

Instructions for developers:
* For each checklist item, if it is N/A to your PR check the N/A box
* For each item that you have done and confirmed for yourself, check Developer box (including if you have checked the N/A box)

Instructions for reviewers:
* For each checklist item that has been confirmed by the Developer, check the Reviewer box if you agree
* For multiple reviewers, feel free to add your own checkbox with your github username next to it if that helps with review tracking

### Code Style

- No deprecated methods are used
  - [x] Developer
  - [x] Reviewer

- No magic strings/numbers - all strings are in `constants` or `messages` files
  - [x] Developer
  - [x] Reviewer
  
- ES queries are wrapped in a Query object rather than inlined in the code
  - [x] Developer
  - [x] Reviewer
  
- Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
  - [x] Developer
  - [x] Reviewer
  
- Cleaned up commented out code, etc
  - [ ] Developer
  - [x] Reviewer

### Testing

- Unit tests have been added/modified
  - [x] Developer
  - [ ] Reviewer
  
- Functional tests have been added/modified
  - [ ] Developer
  - [x] Reviewer
  
- Code has been run manually in development, and functional tests followed locally
  - [x] Developer
  - [ ] Reviewer

### Documentation

- FeatureMap annotations have been added
  - [x] Developer
  - [x] Reviewer
  
- Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
  - [ ] Developer
  - [x] Reviewer
  
- Core model documentation has been added to if needed: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
  - [ ] Developer
  - [x] Reviewer

- Events and consumers documentation has been added if needed: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit
  - [ ] Developer
  - [x] Reviewer
  
- The docs for this branch have been generated and pushed to the doc site (see docs/README.md for details)
  - [ ] Developer
  - [x] Reviewer


### Release Readiness

- If needed, migration has been created and tested locally
  - [ ] Developer
  - [x] Reviewer

- Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit
  - [ ] Developer
  - [ ] Reviewer

- There has been a recent merge up from `develop` (or other base branch).  List the dates of the merges up from develop below
  - 08/04/2024


## Testing

List the Functional Tests that must be run to confirm this feature

We need to do the following tests on the test server before release to production:

1. Confirm an Anon Import is working
2. Run a task to check the latest ES backup
3. Run a task to request an ES backup


## Deployment

### Configuration changes

**N/A**

### Scripts
**N/A**

### Migrations

**N/A**

### Monitoring

**N/A**

### New Infrastructure
**N/A**

### Continuous Integration

CI will no longer need to checkout esprit dependency. Handled by changes in `setup.py`.
